xwiki-rendering-macros/xwiki-rendering-macro-raw/src/main/java/org/xwiki/rendering/internal/macro/raw/RawMacro.java

```diff
@@ -75,8 +84,21 @@ public boolean supportsInlineMode()
 
     @Override
     public List<Block> execute(RawMacroParameters parameters, String content, MacroTransformationContext context)
+        throws MacroExecutionException
     {
         RawBlock rawBlock = new RawBlock(content, parameters.getSyntax());
+
+        try {
+            RawBlockFilterParameters filterParameters = new RawBlockFilterParameters(context);
+            List<RawBlockFilter> filters = this.componentManager.getInstanceList(RawBlockFilter.class);
+            filters.sort((filter1, filter2) -> filter1.getPriority() - filter2.getPriority());
```

Comment:
Same as in the HTML macro: `Comparator.comparingInt` would be more readable. Further, maybe we should indeed remove this code duplication by introducing a helper method, in particular with the sorting step?

Reply:
> Further, maybe we should indeed remove this code duplication by introducing a helper method, in particular with the sorting step?

I'm actually wondering how much all that shouldn't be a default feature for our components in the future. I think we have already a ticket for the order of components. 

But in the meantime I think I'll indeed extract this so that it's easier to change if we need to.